### PR TITLE
Stop providing a hardcoded CA bundle

### PR DESCRIPTION
**Pull request by maxbelanger**
The SDK no longer provides a CA bundle to verify SSL connections. This also allows us to remove the runtime dependency on `pkg_resources` and thus `setuptools`. 

The `ca_certs` parameter is still supported, so users can pin with their own CA bundle if they so choose. Otherwise, the default verification mechanism in the `requests` library now applies (this uses `certifi` and/or system certificates, depending on the configuration).

Improves integration tests to cover both scenarios (i.e. when a bundle is provided, and when one isn't).

## **Checklist**

**General Contributing**
- [X] Have you read the Code of Conduct and signed the [CLA](https://opensource.dropbox.com/cla/)?

**Is This a Code Change?**
- [X] SDK Code Change
- [x] Example/Test Code Change

**Validation**
- [X] Does `tox` pass?
- [X] Do the tests pass?

### CODE REVIEW

File: dropbox/dropbox_client.py [hunk 1]
```diff
@@ -590,7 +590,6 @@ def request_json_string(self,
                                headers=headers,
                                data=body,
                                stream=stream,
-                               verify=True,
```
Comment (maxbelanger) [inline dropbox/dropbox_client.py:593]: Note: this is unnecessary due to setting `session.verify` in `dropbox.session`.

**Review by sderickson — APPROVED** [review]
Looks good, thanks Max!

File: dropbox/session.py [hunk 2]
```diff
@@ -59,8 +51,19 @@ def init_poolmanager(self, connections, maxsize, block=False, **_):
...
-    _session.mount('https://', http_adapter)
 
+    # requests
+    if ca_certs is not None:
+        _session.verify = ca_certs
+    else:
+        _session.verify = True
+
+    # urllib3 within requests
+    http_adapter = _SSLAdapter(pool_connections=4, pool_maxsize=pool_maxsize, ca_certs=ca_certs)
```
Comment (april) [inline dropbox/session.py:65]: We may as well set that to 6, as that's the current number supported by Chrome.

File: test/integration/test_dropbox.py [hunk 3]
```diff
@@ -65,34 +67,43 @@ def _value_from_env_or_die(env_name):
...
 
 @pytest.fixture()
-def refresh_dbx_from_env():
+def refresh_dbx_from_env(dbx_session):
     refresh_token = _value_from_env_or_die(format_env_name(SCOPED_KEY, USER_KEY, REFRESH_TOKEN_KEY))
     app_key = _value_from_env_or_die(format_env_name(SCOPED_KEY, USER_KEY, CLIENT_ID_KEY))
     app_secret = _value_from_env_or_die(format_env_name(SCOPED_KEY, USER_KEY, CLIENT_SECRET_KEY))
     return Dropbox(oauth2_refresh_token=refresh_token,
-                   app_key=app_key, app_secret=app_secret)
+                   app_key=app_key, app_secret=app_secret,
```
Comment (april) [inline test/integration/test_dropbox.py:91]: nit: I'd split these over two lines

File: test/integration/test_dropbox.py [hunk 4]
```diff
@@ -310,3 +291,48 @@ def test_clone_when_team_linked(self, dbx_team_from_env):
...
+        '/oauth2/authorize', host=session.WEB_HOST
+    ) == 'https://{}/oauth2/authorize'.format(session.WEB_HOST)
+
+def test_bad_auth(dbx_session):
+    # Test malformed token
+    malformed_token_dbx = Dropbox(MALFORMED_TOKEN, session=dbx_session)
+    # TODO: backend is no longer returning `BadInputError`
+    # with pytest.raises(BadInputError,) as cm:
+    #     malformed_token_dbx.files_list_folder('')
+    # assert 'token is malformed' in cm.value.message
```
Comment (april) [inline test/integration/test_dropbox.py:318]: Can we remove these comments, or is the intention to fix this?

**Review by april — APPROVED** [review]
Just a few minor changes

File: dropbox/session.py (hunk 2 above)
Comment (maxbelanger) [reply dropbox/session.py:65]: I'd prefer not as it's not really related to the CA bundle changes being made here. We should do that separately IMO.

File: test/integration/test_dropbox.py (hunk 4 above)
Comment (maxbelanger) [reply test/integration/test_dropbox.py:318]: The intention is to fix this in a future PR. This is also not technically introduced in this PR, I've just moved this test out of the class.

File: dropbox/session.py (hunk 2 above)
Comment (april) [reply dropbox/session.py:65]: I see, I thought it was new code but maybe the `pool_connections=4` was carried over from somewhere else?

**Pull request merged**